### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Kozubowicz**

### CODE REVIEW

File: src/fillTank.test.js [hunk 1]
```diff
@@ -1,11 +1,147 @@
...
+    fillTank(customer, 5);
+
+    expect(customer).toEqual({
+      money: 2850,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 40,
+      },
+    });
+  });
```
Comment (zibi-dev) [inline src/fillTank.test.js]: The test named should ordered full tank if amount is not given does not actually handle the "full tank" scenario if no amount is passed. It calls `fillTank(customer, 5)`

File: src/fillTank.test.js [hunk 2]
```diff
@@ -1,11 +1,147 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should ', () => {});
```
Comment (zibi-dev) [inline src/fillTank.test.js]: remove it

File: src/fillTank.test.js [hunk 3]
```diff
@@ -1,11 +1,147 @@
...
 
+  // write tests here
```
Comment (zibi-dev) [inline src/fillTank.test.js]: clear comments

File: src/fillTank.test.js [hunk 4]
```diff
@@ -1,11 +1,147 @@
...
+    fillTank(customer, 5, 50);
+
+    expect(customer).toEqual({
+      money: 2850,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 40,
+      },
+    });
+  });
```
Comment (zibi-dev) [inline src/fillTank.test.js]: In the test it should not pour more than tank can accommodate, you are passing a third parameter, 50, as the amount to be poured. However, the third parameter isn't explained in the test description or logic.

File: src/fillTank.test.js [hunk 5]
```diff
@@ -1,11 +1,147 @@
...
+    fillTank(customer, 5.12, 4.6);
+
+    expect(customer).toEqual({
+      money: 2976.45,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 12.6,
+      },
+    });
+  });
```
Comment (zibi-dev) [inline src/fillTank.test.js]: Multiple test cases are similar or overlapping in their goals. For example, should round the poured amount by discarding number to the tenth part and should round the price of the purchased fuel to the nearest hundredth part could be combined into a more holistic test if the goal is to check rounding behavior.

File: src/fillTank.test.js [hunk 6]
```diff
@@ -1,11 +1,147 @@
...
+  it(`should not fill more than client can pay`, () => {
+    const customer = {
+      money: 50,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 10,
+      },
+    };
+
+    fillTank(customer, 5, 20);
```
Comment (zibi-dev) [inline src/fillTank.test.js]: Use variables to represent reusable values like the price per liter, tank amount, and so on. This helps in maintaining the code and reduces hardcoding.

```suggestion
      const fuelPrice= 5;
      const fuelAmount= 40;
      
      fillTank(customer, fuelPrice, fuelAmount);
```

**Review by Zibi95 — CHANGES_REQUESTED** [review]
Hey, I've left some comments to improve the tests.

File: src/fillTank.test.js (hunk 1 above)
Comment (Kozubowicz) [reply src/fillTank.test.js]: I do not understand you, default value for amount is Infinity

**Review by natalia-klonowska — CHANGES_REQUESTED** [review]
Great 😃 just one last thing to change: combine the tests that check if rounding works properly. and add test that checks negative numbers

**Review by Zibi95 — APPROVED** [review]
Good job!
